packages/hash/api/src/model/knowledge/entity.model.ts

```diff
@@ -112,6 +118,139 @@ export default class {
     });
   }
 
+  /**
+   * Create an entity along with any new/existing entities specified through links.
+   *
+   * @param params.createdById the account id that is creating the entity
+   * @param params.entityDefinition the definition of how to get or create the entity optionally with linked entities
+   */
+  static async createEntityWithLinks(
+    graphApi: GraphApi,
+    params: {
+      createdById: string;
+      entityDefinition: KnowledgeEntityDefinition;
+    },
+  ): Promise<EntityModel> {
+    const { createdById, entityDefinition: entityDefinitions } = params;
+
+    const entitiesInTree = linkedTreeFlatten<
+      KnowledgeEntityDefinition,
+      KnowledgeLinkedEntityDefinition,
+      "linkedEntities",
+      "entity"
+    >(entityDefinitions, "linkedEntities", "entity");
+
+    const entities = await Promise.all(
+      entitiesInTree.map(async (entityDefinition) => ({
+        link: entityDefinition.meta
+          ? {
+              parentIndex: entityDefinition.parentIndex,
+              meta: entityDefinition.meta,
+            }
+          : undefined,
+        entity: await EntityModel.getOrCreate(graphApi, {
+          createdById,
+          entityDefinition,
+        }),
```

Comment:
Once the Graph API supports validating for required links, we may have to change the strategy here so that entities are created with their initial links within a single request.

Just wanted to flag this, perhaps worth leaving a todo but it will also be obvious if we implement required link validation.

Reply:
Yes very true, I've created an asana ticket and also linked it with a todo in b98aee99cdc65c9758f085598f2c50097b903c86